### PR TITLE
[#232] Introduce document highlighting

### LINKED ISSUE #232: implement textDocument/documentHighlight
We currently have an implementation of [`textDocument/references`](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_references).

Use this as a basis for [`textDocument/documentHighlight`](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_documentHighlight)

### PR DESCRIPTION
**Pull request by alanz**
### Description

Adds support for document highlighting.

This commit introduces the feature.  There will be follow-up ones to improve the quality of the highlighting provided.

Fixes #232 .

### CODE REVIEW

File: src/els_references_provider.erl [hunk 1]
```diff
@@ -64,3 +77,35 @@ location(Uri, Range) ->
...
+      R
+  end;
+find_highlights(_Uri, _POI) ->
+  null.
+
+-spec document_highlight(poi_range()) -> map().
+document_highlight(Range) ->
+    %% TODO:AZ:range as displayed ends one col too early
+  #{ range => els_protocol:range(Range)
+   , kind => 1
```
Comment (jfacorro) [inline src/els_references_provider.erl]: `erlang_ls.hrl` has these defined as macros.

File: src/els_methods.erl [hunk 2]
```diff
@@ -110,6 +111,8 @@ initialize(Params, State) ->
                }
           , definitionProvider => els_definition_provider:is_enabled()
           , referencesProvider => els_references_provider:is_enabled()
+            %% AZ:TODO: different enable flag for documentHighlightProvider
+          , documentHighlightProvider => els_references_provider:is_enabled()
```
Comment (jfacorro) [inline src/els_methods.erl]: Why not create a new `els_document_highlight_provider` and move the code added to `els_references_provider` there?

Comment (alanz) [reply src/els_methods.erl]: I was not sure initially how much overlap there would be, but I presumed that would be the eventual outcome. Will do it now.

**Comment by alanz** [comment]
I force-pushed with only an update to the commit description, and it gives a failure in a different test.

Any ideas?

**Comment by jfacorro** [comment]
I  think it was just the flakiness we sometimes see in the proper tests. I reried the build and it now succeeds.

**Comment by robertoaloi** [comment]
Nice one. Thanks @alanz !

**Pull request merged**